src/widget/findondiscogs.cpp

```diff
@@ -0,0 +1,37 @@
+#include "findondiscogs.h"
+
+#include <QMenu>
+#include <QtDebug>
+
+#include "track/track.h"
+#include "util/parented_ptr.h"
+
+namespace {
+const QString kServiceTitle = QStringLiteral("Discogs");
+
+const QString kSearchUrl = QStringLiteral(
+        "https://www.discogs.com/search/?"); // List of Query needed.
+} //namespace
+
+void FindOnDiscogs::addSubmenusForServices(QMenu* pFindOnMenu, const Track& track) {
+    const auto artist = track.getArtist();
+    const auto trackTitle = track.getTitle();
+    const auto album = track.getAlbum();
+    auto m_pServiceMenu = make_parented<QMenu>(this);
```

Comment:
I think you can use "this" directly, because this is already a QMenu.

Reply:
I have used `FindOnSoundcloud* m_pServiceMenu = this;`